src/data/mystery-encounters/encounters/dark-deal-encounter.ts

```diff
@@ -160,7 +162,13 @@ export const DarkDealEncounter: MysteryEncounter =
           scene.unshiftPhase(new ModifierRewardPhase(scene, modifierTypes.ROGUE_BALL));
 
           // Start encounter with random legendary (7-10 starter strength) that has level additive
-          const bossTypes: Type[] = encounter.misc.removedTypes;
+          // If this is a mono-type challenge, always ensure the required type is filtered for
+          let bossTypes: Type[] = encounter.misc.removedTypes;
+          const singleTypeChallenges = scene.gameMode.challenges.filter(c => c.id === Challenges.SINGLE_TYPE);
+          if (scene.gameMode.isChallenge && singleTypeChallenges.length > 0) {
+            bossTypes = singleTypeChallenges.map(c => c.value as Type);
```

Comment:
```suggestion
          const singleTypeChallenges = scene.gameMode.challenges.filter(c => c.value && c.id === Challenges.SINGLE_TYPE);
          if (scene.gameMode.isChallenge && singleTypeChallenges.length > 0) {
            bossTypes = singleTypeChallenges.map(c => (c.value - 1) as Type);
```

There is an off-by-one error here because a `c.value' of 0 is used for "no challenge". The following screenshot is from a mono-Water challenge:
![20241021 Dark Deal](https://github.com/user-attachments/assets/eefdceb0-5695-46f2-bfee-216d2518d18e)